### DISCUSSION TITLE
How to Cast a EntityObject to an specific Entity like LwPolylineVertex?

### DISCUSSION THREAD

**Discussion by Jan21021** [haplokuon/netDxf#269] [Q&A] [answered]
Hey, I am sorry for my dumb beginner question but i want to get the coordinates from an existing BoundaryPath. 

I easy get all the Entity Objects with something like this: 

foreach (var hatch in dxf.Hatches)
{
    var boundarys = hatch.BoundaryPaths[0];
    var entities = boundarys.Entities;
}

but I am stuck how to cast the EntityObjects to the specific entity like a polyline =(
Help would be very much appreciated =)

Absolute awesome nugget by the way. Very cool and easy to use!

**Comment by haplokuon** [comment, marked as answer]
Casting an EntityObject to the specific type of entity is done like any other casting, there is nothing special about it as long as you are casting your object to the correct class that inherits from it. You cannot do what the question in your title suggest "How to Cast a EntityObject to an specific Entity like LwPolylineVertex?", the LwPolylineVertex class does not inherit from EntityObject. You can find more specific instructions on how casting is done in the documentation of the C# language.
```c#
EntityObject entity;

LwPolyline p1 = (LwPolyline) entity; // this is not safe since multiple entity classes inherit from EntityObject

// perform a safe cast, there are several methods the syntax may depend on the C# language version you are working with
LwPolyline p2 = entity as LwPolyline;
if (entity != null)
{
	// do something with p2
}

// or
if (entity is LwPolyline p3)
{
	// do something with p3
}

// additionally, the EntityObject class contains a property called Type that will tell you which kind of entity the object represents.
if (entity.Type == EntityType.LwPolyline)
{
	LwPolyline p4 = (LwPolyline) entity; // now you can safely cast the object "entity" since you know its type
	// do something with p4
}
```
Post a sample code pointing out which is the entity object you are trying to cast.

And now a couple of comments on your post.
1. In your code "boundarys.Entities" is a list, you still need to iterate through the items in the collection. Every time you see a class property as a plural is a collection not a single item.
2. If you want to know the actual perimeter of your hatch you need to read the Edges property of the HatchBoundaryPath. A hatch can have multiple boundaries, for example holes in a larger area, and each boundary can be made of multiple edges. If the hatch is associative it will also contain the entities associated to that specific boundary in the Entities collection, but they may not define a closed shape, you can read more about it [here](https://github.com/haplokuon/netDxf/issues/184). 
3. I do not upload anything about netDxf outside Github.

**Reply by Jan21021** [reply]
hey thank you very much for your fast replay! thats very cool to have so fast an answer from the creater of a nugget =) 
i made a really dumb mistake by trying to cast the entity as LwPolylineVertex, i only tried to cast that, waht didn't work so i never even tried to cast the entity to a LwPolyline..... yes yes shame on me. 

1. got it
2. thanks for that input
3. ok

thank you again for your help and your awesome nugget!
